test/parallel/xyk-pallet.API.poolLiqudity.sepClaimRewards.test.ts

```diff
@@ -0,0 +1,144 @@
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  [testUser, testUser1, testUser2] = setupUsers();
+
+  await setupApi();
+
+  [token] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue, defaultCurrencyValue, defaultCurrencyValue],
+    sudo
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintToken(token, testUser, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser),
+    Assets.mintToken(token, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1),
+    Assets.mintToken(token, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+});
+
+test("One user claim all the rewards on every session and other user claim them at the 3rd session, the sum of rewards are equal", async () => {
+  liqIdPromPool = await getLiquidityAssetId(MGA_ASSET_ID, token);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqIdPromPool.toNumber(), 20)
+  );
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(MGA_ASSET_ID, token, defaultCurrencyValue)
+    ),
+    Sudo.sudoAs(
+      testUser2,
+      Xyk.mintLiquidity(MGA_ASSET_ID, token, defaultCurrencyValue)
+    )
+  );
+
+  await waitForRewards(testUser1, liqIdPromPool);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqIdPromPool)),
+    Sudo.sudoAs(testUser2, Xyk.claimRewardsAll(liqIdPromPool))
+  );
+
+  const rewardsInfoUser1Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdPromPool
+  );
+
+  const rewardsInfoUser2Before = await getRewardsInfo(
+    testUser2.keyRingPair.address,
+    liqIdPromPool
+  );
+
+  for (let index = 0; index < 2; index++) {
+    await waitForRewards(testUser1, liqIdPromPool);
+
+    await claimRewardsAll(testUser1, liqIdPromPool);
+  }
+
+  await waitForRewards(testUser1, liqIdPromPool);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqIdPromPool)),
+    Sudo.sudoAs(testUser2, Xyk.claimRewardsAll(liqIdPromPool))
+  );
+
+  const rewardsInfoUser1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdPromPool
+  );
+
+  const rewardsInfoUser2After = await getRewardsInfo(
+    testUser2.keyRingPair.address,
+    liqIdPromPool
+  );
+  const rewardsClaimedUser1 = rewardsInfoUser1After.rewardsAlreadyClaimed.sub(
+    rewardsInfoUser1Before.rewardsAlreadyClaimed
+  );
+  const rewardsClaimedUser2 = rewardsInfoUser2After.rewardsAlreadyClaimed.sub(
+    rewardsInfoUser2Before.rewardsAlreadyClaimed
+  );
+
+  expect(rewardsClaimedUser1).bnEqual(rewardsClaimedUser2);
```

Comment:
can you plz check if the storage are almost identical? I think they should be.

Reply:
Done